test-specifications/token-service/tokenCreateTransaction.md

```diff
@@ -0,0 +1,1150 @@
+| 5       | Creates a token with no symbol                                | name="testname", treasuryAccount=<OPERATOR_ACCOUNT_ID>                                                                                                                 | The token creation fails with a MISSING_TOKEN_SYMBOL response code from the network.                                                                              | N                 |
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 641,
+  "method": "createToken",
+  "params": {
+    "name": "testname",
+    "symbol": "testsymbol",
+    "treasuryAccount": "0.0.2"
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 641,
+  "result": {
+    "tokenId": "0.0.836",
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Decimals:**
+
+- The number of decimal places by which a token is divisible.
+
+| Test no | Name                                                                           | Input                                                                                                    | Expected response                                                                       | Implemented (Y/N) |
+|---------|--------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates a fungible token with a valid amount of decimals                       | name="testname", symbol="testsymbol", decimals=3, treasuryAccount=<OPERATOR_ACCOUNT_ID>                  | The token creation succeeds and the token has 3 decimals.                               | N                 |
+| 2       | Creates a fungible token with the minimum amount of decimals                   | name="testname", symbol="testsymbol", decimals=0, treasuryAccount=<OPERATOR_ACCOUNT_ID>                  | The token creation succeeds and the token has 0 decimals.                               | N                 |
+| 3       | Creates a fungible token with a decimal amount below the minimum amount        | name="testname", symbol="testsymbol", decimals=-1, treasuryAccount=<OPERATOR_ACCOUNT_ID>                 | The token creation fails with an INVALID_TOKEN_DECIMALS response code from the network. | N                 |
+| 4       | Creates a fungible token with the maximum amount of decimals                   | name="testname", symbol="testsymbol", decimals=2147483647, treasuryAccount=<OPERATOR_ACCOUNT_ID>         | The token creation succeeds and the token has 2,147,483,647 decimals.                   | N                 |
```

Comment:
We should verify this, there _may_ be an actual limit of 2<sup>31</sup> because of Java limitations on the consensus node.